codexctl/analysis.py

```diff
@@ -31,3 +35,108 @@ def get_update_image(file: str):
         warnings.warn("Unable to open public key", RuntimeWarning)
 
     return image, volume
+
+
+def get_swu_metadata(swu_file: str) -> Tuple[str, HardwareType]:

```

Comment:
```suggestion
def get_swu_metadata(swu_file: str) -> tuple[str, HardwareType]:
```

As explained in other comments, this should not be done in codexctl, and the minor enhancement to remarkable-update-image to provide which device the swu file is for should be implemented instead.